### ISSUE TITLE
[Bug] Fastlane Produce when app_identifier is set to an array.

### ISSUE THREAD

**Issue by justMaku** [fastlane/fastlane#12982]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
Since our app also offers Apple Watch extension, we have set `app_identifier` in `Appfile` to an array:
```
app_identifier [
    "target.iphone", 
    "target.watch", 
    "target.watchextension"
]
```

This works fine with `fastlane match` but unfortunately it fails when used with `fastlane produce` as shown below.

I have redacted some of the output from the `fastlane env` command (namely Fastfile), let me know if there's something you might need so I can share it privately.

##### Complete output when running fastlane, including the stack trace and command used
<details>
  <pre>
> bundle exec fastlane produce                 
[✔] 🚀 

+----------------+------------------------------------------------------------------+
|                            Summary for produce 2.100.1                            |
+----------------+------------------------------------------------------------------+
| username       | ios@somecompany.com                                              |
| app_identifier | ["target.iphone", "target.watch", "target.watchextension"] |
| sku            | 1532955410                                                       |
| platform       | ios                                                              |
| language       | English                                                          |
| skip_itc       | false                                                            |
| skip_devcenter | false                                                            |
| team_id        | SOME_TEAM_ID                                                       |
+----------------+------------------------------------------------------------------+

[14:56:54]: To not be asked about this value, you can specify it using 'app_name'
[14:56:54]: App Name: SomeName
[14:56:58]: Creating new app 'SomeName' on the Apple Dev Center

Looking for related GitHub issues on fastlane/fastlane...

[!] Apple provided the following error info:
	'"target.iphone", "target.watch", "target.watchextension"' is not a valid identifier for an app.  Please choose an identifier containing only alphanumeric characters, dots, asterisk and that does not start with pass..
</pre>
</details>

### Environment
<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.14                                                          |
| Ruby                        | 2.3.7                                                          |
| Bundler?                    | true                                                           |
| Git                         | git version 2.17.1                                             |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Mac OS X 10.14 (18A336e)                                       |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib |
| OpenSSL Version             | LibreSSL 2.6.4                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode-beta.app/Contents/Developer/               |
| Xcode Version               | 10.0                                                           |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
#REDACTED
```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
apple_id "ios@some-company.com"
team_id "SOME_TEAM_ID"

app_identifier [
    "target.iphone", 
    "target.watch", 
    "target.watchextension"
]

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.100.1 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin               | Version | Update-Status |
| -------------------- | ------- | ------------- |
| fastlane-plugin-brew | 0.1.1   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.0        |
| bundler                   | 1.16.3       |
| io-console                | 0.4.5        |
| CFPropertyList            | 3.0.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.2        |
| atomos                    | 0.1.2        |
| babosa                    | 1.0.2        |
| claide                    | 1.0.2        |
| colored                   | 1.2          |
| colored2                  | 3.1.2        |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| unf_ext                   | 0.0.7.5      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20180417 |
| dotenv                    | 2.5.0        |
| emoji_regex               | 0.1.1        |
| excon                     | 0.62.0       |
| multipart-post            | 2.0.0        |
| faraday                   | 0.15.2       |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.12.2       |
| fastimage                 | 2.1.3        |
| gh_inspector              | 1.1.3        |
| jwt                       | 1.5.6        |
| little-plugger            | 1.1.4        |
| multi_json                | 1.13.1       |
| logging                   | 2.2.2        |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| signet                    | 0.8.1        |
| googleauth                | 0.6.2        |
| httpclient                | 2.8.3        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| google-api-client         | 0.21.2       |
| json                      | 2.1.0        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| plist                     | 3.4.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.5        |
| slack-notifier            | 2.3.2        |
| terminal-notifier         | 1.8.0        |
| unicode-display_width     | 1.4.0        |
| terminal-table            | 1.8.0        |
| tty-screen                | 0.6.5        |
| tty-cursor                | 0.6.0        |
| tty-spinner               | 0.8.0        |
| word_wrap                 | 1.0.0        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.5.9        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| xcpretty-travis-formatter | 1.0.0        |
| fastlane-plugin-brew      | 0.1.1        |
</details>
*generated on:* **2018-07-30**
</details>

**Comment by joshdholtz** [comment]
@justMaku Heyyy 👋 

1) Thanks for making the PR the other day for filtering out the extra files inside of an Xcode project 😊 
2) The `app_identifier` in _produce_ needs to be a single value. It is used to create the single main "App ID" like you would inside of the iOS developer portal. If you are using watch and watch extension, I think you are looking to create and associate an app group? https://docs.fastlane.tools/actions/produce/#enabling-disabling-application-services

```sh
fastlane produce group -g group.krausefx -n "Example App Group"
fastlane produce associate_group -a com.krausefx.app group.krausefx
```

🤔

**Comment by justMaku** [comment]
I think I figured out what's going on: I only had an `Appfile` for my fastlane setup and didn't use `deliver` or `pilot` previously, just `match` for managing provisioning profiles for my targets. Without `Matchfile`, `match` takes the `app_identifier` value from the `Appfile` and accepts arrays even though, technicaly the `app_identifier` in `Appfile` should be a `string`. When no `Deliverfile` was present, `deliver` was getting it `app_identifier` value also from `Appfile` but since it expects a `string` there was an issue. I fixed this by creating `Matchfile` with an array `app_identifier`, `Deliverfile` with `string` value for `app_identifier` (just main target) and everything is working fine now.
